csharp/src/Ice/Communicator.cs

```diff
@@ -360,7 +373,126 @@ private class ObserverUpdater : Instrumentation.IObserverUpdater
+
+                string val = GetProperty("Ice.Default.EndpointSelection") ?? "Random";
+                if (val.Equals("Random"))
+                {
+                    DefaultEndpointSelection = Ice.EndpointSelectionType.Random;
+                }
+                else if (val.Equals("Ordered"))
+                {
+                    DefaultEndpointSelection = EndpointSelectionType.Ordered;
+                }
+                else
+                {
+                    throw new InvalidConfigurationException($"illegal value `{val}'; expected `Random' or `Ordered'");
+                }
+
+                DefaultTimeout = GetPropertyAsInt("Ice.Default.Timeout") ?? 60000;
+                if (DefaultTimeout < 1 && DefaultTimeout != -1)
+                {
+                    Logger.Warning($"invalid value for Ice.Default.Timeout `{DefaultTimeout}': defaulting to 60000");
+                    DefaultTimeout = 60000;
+                }
+
+                DefaultLocatorCacheTimeout = GetPropertyAsInt("Ice.Default.LocatorCacheTimeout") ?? -1;
+                if (DefaultLocatorCacheTimeout < -1)
+                {
+                    Logger.Warning(
+                    $"invalid value for Ice.Default.LocatorCacheTimeout `{DefaultLocatorCacheTimeout}': defaulting to -1");
+                    DefaultLocatorCacheTimeout = -1;
+                }
+
+                DefaultInvocationTimeout = GetPropertyAsInt("Ice.Default.InvocationTimeout") ?? -1;
+                if (DefaultInvocationTimeout < 1 && DefaultInvocationTimeout != -1)
+                {
+                    Logger.Warning(
+                        $"invalid value for Ice.Default.InvocationTimeout `{DefaultInvocationTimeout}': defaulting to -1");
+                    DefaultInvocationTimeout = -1;
+                }
+
+                // TODO: switch to 0 default
+                DefaultPreferNonSecure = (GetPropertyAsInt("Ice.Default.PreferNonSecure") ?? 1) > 0;
+
+                if (GetProperty("Ice.Default.Encoding") is string encoding)
+                {
+                    DefaultEncoding = Encoding.Parse(encoding);
+                    DefaultEncoding.CheckSupported();
+                }
+                else
+                {
+                    DefaultEncoding = Encoding.Latest;
+                }
+
+                DefaultFormat = (GetPropertyAsInt("Ice.Default.SlicedFormat") > 0) ? Ice.FormatType.Sliced
+                                                                                   : Ice.FormatType.Compact;
+
+                {
+                    if (GetPropertyAsInt("Ice.Override.Timeout") is int timeout)
+                    {
+                        OverrideTimeout = timeout;
+                        if (timeout < 1 && timeout != -1)
+                        {
+                            Logger.Warning($"invalid value for Ice.Override.Timeout `{timeout}': defauling to -1");
```

Comment:
I think your previous comment is correct, and that instead of warning, we should throw an `InvalidConfigurationException` instead. If we switch to this behavior, it'll no longer be an issue.